liferay-tango/liferay-portal
# LPS-177730 Migrate IntegerInput component to Typescript

Motivation
=======
The motivation of this task is to change the Integer Input from JavaScript to Typescript and making the component functional. This is part of an initiative to migrate our Segments Editor components to Typescript.

[Link to story or ticket](https://issues.liferay.com/browse/LPS-177657)
[Link to technical task](https://issues.liferay.com/browse/LPS-177730)

Proposed Solution
========
The solution consists in delete the IntegerInput.es.js component and create an IntegerInput.tsx and change the structure of the file to make it functional.

Steps to Verify:
----------------
1. Create an Event Segment in AC
2. Go to Control Panel > Job Scheduler > Scheduled Jobs
3. Run CheckIndividualSegmentsMessageListener
4. Go to Control Panel > Users > Segmentation
5. Edit a Document & Media Segment
6. Check that the Integer Input works correctly by changing the value and moving the focus out of the input
